Sources/SwiftParser/Specifiers.swift

```diff
@@ -596,12 +582,24 @@ extension TokenConsumer {
 // MARK: - Parsing effect specifiers
 
 extension Parser {
-  private mutating func parseThrownTypeClause() -> RawThrownTypeClauseSyntax {
-    let (unexpectedBeforeLeftParen, leftParen) = self.expect(.leftParen)
+  /// Parse a throws clause after we've already parsed the 'throws' keyword to introduce it.
+  private mutating func parseThrowsClause(after throwsKeyword: RawTokenSyntax) -> RawThrowsClauseSyntax {
+    guard self.at(.leftParen) && experimentalFeatures.contains(.typedThrows) else {
+      return RawThrowsClauseSyntax(
+        throwsSpecifier: throwsKeyword,
+        leftParen: nil,
+        type: nil,
+        rightParen: nil,
+        arena: self.arena
+      )
+    }
+
+    let (unexpectedBetweenThrowsSpecifierAndLeftParen, leftParen) = self.expect(.leftParen)
```

Comment:
Instead of doing `self.at(.leftParen)` above and then `self.expect`, you can just do the following above.

```swift
guard experimentalFeatures.contains(.typedThrows), let leftParenn = self.consume(if: .leftParen)
```